vscode-trace-extension/src/extension.ts

```diff
@@ -162,6 +168,7 @@ export function activate(context: vscode.ExtensionContext): ExternalAPI {
                 return;
             }
             await startTraceServerIfAvailable(traceUri.fsPath);
+            await updateUris();
```

Comment:
same comment here. The URL should not be updated everytime a trace is opened. It should be only be done if necessary.